src/renderer/components/+workspaces/workspaces.tsx

```diff
@@ -1,216 +1,89 @@
+        options.push({ value: ChooseWorkspace.editActionId, label: "Edit current workspace ..." });
+      }
+    }
 
-  renderInfo() {
-    return (
-      <Fragment>
-        <h2>What is a Workspace?</h2>
-        <p className="info">
-          Workspaces are used to organize number of clusters into logical groups.
-        </p>
-        <p>
-          A single workspaces contains a list of clusters and their full configuration.
-        </p>
-      </Fragment>
-    );
+    return options;
   }
 
-  saveWorkspace = (id: WorkspaceId) => {
-    const workspace = new Workspace(this.editingWorkspaces.get(id));
-
-    if (workspaceStore.getById(id)) {
-      workspaceStore.updateWorkspace(workspace);
-      this.clearEditing(id);
+  onChange(id: string) {
+    if (id === ChooseWorkspace.addActionId) {
+      CommandOverlay.open(<AddWorkspace />);
 
       return;
     }
 
-    if (workspaceStore.addWorkspace(workspace)) {
-      this.clearEditing(id);
-    }
-  };
+    if (id === ChooseWorkspace.removeActionId) {
+      CommandOverlay.open(<RemoveWorkspace />);
 
-  addWorkspace = () => {
-    const workspaceId = uuid();
-
-    this.editingWorkspaces.set(workspaceId, new Workspace({
-      id: workspaceId,
-      name: "",
-      description: ""
-    }));
-  };
+      return;
+    }
 
-  editWorkspace = (id: WorkspaceId) => {
-    const workspace = workspaceStore.getById(id);
+    if (id === ChooseWorkspace.editActionId) {
+      CommandOverlay.open(<EditWorkspace />);
 
-    this.editingWorkspaces.set(id, toJS(workspace));
-  };
+      return;
+    }
 
-  activateWorkspace = (id: WorkspaceId) => {
+    workspaceStore.setActive(id);
     const clusterId = workspaceStore.getById(id).lastActiveClusterId;
 
-    workspaceStore.setActive(id);
-    clusterStore.setActive(clusterId);
-  };
-
-  clearEditing = (id: WorkspaceId) => {
-    this.editingWorkspaces.delete(id);
-  };
-
-  removeWorkspace = (id: WorkspaceId) => {
-    const workspace = workspaceStore.getById(id);
-
-    ConfirmDialog.open({
-      okButtonProps: {
-        label: `Remove Workspace`,
-        primary: false,
-        accent: true,
-      },
-      ok: () => {
-        this.clearEditing(id);
-        workspaceStore.removeWorkspace(workspace);
-      },
-      message: (
-        <div className="confirm flex column gaps">
-          <p>
-            Are you sure you want remove workspace <b>{workspace.name}</b>?
-          </p>
-          <p className="info">
-            All clusters within workspace will be cleared as well
-          </p>
-        </div>
-      ),
-    });
-  };
-
-  onInputKeypress = (evt: React.KeyboardEvent<any>, workspaceId: WorkspaceId) => {
-    if (evt.key == "Enter") {
-      // Trigget input validation
-      evt.currentTarget.blur();
-      evt.currentTarget.focus();
-      this.saveWorkspace(workspaceId);
+    if (clusterId) {
+      navigate(clusterViewURL({ params: { clusterId } }));
```

Comment:
Something odd with workspace creation/navigation is happening. Probably this comment relates more to `AddWorkspace` component.

After creating new workspace - a previous one somehow sticked to it. So when you switch to newly created workspace you'll see some cluster opened. Better show than explain:

https://user-images.githubusercontent.com/9607060/106567226-89424280-6542-11eb-9ef0-292d4a5a61de.mp4

Reply:
Are we sure this behaviour is not in the master branch? I have seen similar behaviour in the past and that's why this PR originally always reseted active cluster to `null` when active workspace changed (but I reverted that because of code review comments).

Reply:
This PR now explicitly sets active cluster to `null` when workspace is added. I feel that we still have a bug somewhere else in workspace switch logic but that probably is a topic for an another PR.

Reply:
I've checked `master` branch. New workspace created without any cluster attached to it.

https://user-images.githubusercontent.com/9607060/106575957-ae887e00-654d-11eb-9ad7-addac55b80d9.mp4